deserializers/src/main/java/com/nextdoor/bender/deserializer/json/GenericJsonDeserializer.java

```diff
@@ -36,21 +41,49 @@
   protected JsonParser parser;
   private final List<FieldConfig> nestedFieldConfigs;
   private String rootNodeOverridePath;
+  private final boolean performBase64DecodeAndUnzip;
 
   public GenericJsonDeserializer(List<FieldConfig> nestedFieldConfigs) {
-    this(nestedFieldConfigs, null);
+    this(nestedFieldConfigs, null, false);
   }
 
   public GenericJsonDeserializer(List<FieldConfig> nestedFieldConfigs,
-      String rootNodeOverridePath) {
+                                 String rootNodeOverridePath,
+                                 boolean performBase64DecodeAndUnzip) {
     this.nestedFieldConfigs = nestedFieldConfigs;
     this.rootNodeOverridePath = rootNodeOverridePath;
+    this.performBase64DecodeAndUnzip = performBase64DecodeAndUnzip;
+  }
+
+  public byte[] readGzipCompressedData(byte[] data) throws IOException {
+    GZIPInputStream gzipInputStream = new GZIPInputStream(new ByteArrayInputStream(data));
+    ByteArrayOutputStream byteArrayOutputStream = new java.io.ByteArrayOutputStream();
```

Comment:
Same thing here. If possible I'd like to see us do this in a streaming manner.

Reply:
Yup this can be used via the reset() method, changing this one to be in the constructor.